M2/Macaulay2/d/boost-regex.cpp

```diff
@@ -0,0 +1,290 @@
+#include "strings-exports.h"
+
+#include <boost/regex.hpp>
+#include <string.h>
+#include <iostream>
+
+#define DEBUG_REGEX 0
+
+/* Keep this enum in sync with RegexFlags in m2/regex.m2 */
+enum RegexFlags {
```

Comment:
This enum is not needed: use boost's `enum flag_type_` instead.

Reply:
We use a bitmask to communicate the flags from top level to the engine, so more than the enum, we need to know the actual numbers so they can be written in regex.m2. However, the specific numbers in boost are internal numbers and are not guaranteed to stay the same from version to version. That's why we can't use boost's own enum. Moreover, boost actually has two separate enums, one for regex flags and one for match flags. If we were to use it, we'd need two flags, which is more than we need.

Reply:
Re:
```m2
-- Keep this enum in sync with RegexFlags in Macaulay2/e/regex.cpp
RegexFlags = new HashTable from {
...
```

Let's not introduce any more pairs of enums that have to be kept in sync, for it will be a future headache.  Instead, create core variables with names such as Icase and values such as 32.  Then you could write `"Icase"=>Icase` in regex.m2

Reply:
That doesn't make sense. Don't you still need to keep the value 32 in sync between regex.m2 and boost-regex.cpp?

Reply:
Oh, I should have added that the variable Icase would be defined in regex.dd, where the enum values are available.

Reply:
Alternatively, the code for translating the string "Icase" to the number 32 could be entirely in regex.dd.

Reply:
You worry about a "future headache" that seems completely imaginary to me. Plenty of programs pass around status codes between different environments like this. There are tests to make sure that the exported and documented options, namely RegexPOSIX and RegexPerl, are kept in sync.

Reply:
Is this still not resolved?